proposals/1/standardized-mod-format.md

```diff
@@ -0,0 +1,128 @@
+# Standardized Mod Format (DRAFT!)
+
+## Description Of Issue
+
+Lack of a clear specification between modloaders on a commonly supported mod structure & format (such as package.json details).
+Right now the featureset that's common between CCLoader and CCInjector isn't clear without reading their code.
+This would define a reliable featureset between these modloaders and any other modloaders that enter the scene.
+
+## Solution
+
+A specification which exactly defines specific terms, and does not define legacy formats from the previous era.
+(Note in regards to this: Defining legacy formats from the previous era would either require support for them or waste space or be completely pointless)
+
+### Advantages
+
+- Modloaders and modders know what is supported everywhere.
+
+- Less likelihood to use outdated mechanisms from the old era
+
+- APIs defined in other specifications may gain reference implementations using this specification.
+
+### Disadvantages
+
+- No specific system in place for per-modloader extensions other than extra unspecified fields (for optional dependency) or Virtual Dependencies (which cause non-compliance and make the mod non-portable).
+
+### Regarding Implementation
+
+It is important to note that some features of this specification may, themselves, be implemented within mods.
+This is fine - the implementation of this specification is then a matter of "that modloader + these mods".
+This in particular applies to how CCLoader handles patch files via `simplify`.
+
+## Mod Format
+
+A mod is in a directory. The directory's name is irrelevant.
+
+The directory then contains a file called "package.json" (see The package.json format).
+
+### The package.json format
+
+The package.json format describes a mod.
+
+It is a JSON file, with an object containing the following keys ('Optional' allows the key to be missing):
+
+```
+{
+ "name": <String, serves as the mod's ID>,
+ "version": <Semver-compliant version string>,
+ "ccmodDependencies": <
+  Optional Object - keys are mod IDs, values are semver-compliant version constraints
+  This is used between different mods to establish load order & required other mods.
+  Note that some entries in here may not actually be mods - see Virtual Dependencies for further information.
+ >,
+ "dependencies": <
+  Optional Object - due to compatibility, this is a bit awkward
+  If ccmodDependencies is provided, keys are Node modules, values are semver-compliant version constraints
+  If ccmodDependencies is NOT provided, this key is effectively renamed to it - meaning keys are mod IDs, not Node modules.
+ >,
+
+ "description": <Optional human-readable string>,
```

Comment:
Request for adding fields `author` (optional; maintainer/main author) and `contributors` (optimal, array). Reference: https://docs.npmjs.com/files/package.json#people-fields-author-contributors

Reply:
Resolved with b5d0e83 clarifying that these are allowed without explicitly defining their format.

Reply:
I would prefer to have `author` and `contributors` standardized. So every tool/loader can use this data if needed.